### ISSUE TITLE
Floating Action Button Enhancements

### ISSUE THREAD

**Issue by BingoRUS** [DevExpress/DevExtreme#9610]
# The Problem

The Floating Action Button does not allow users to:

- Add a label to an action.
- Hide an action.
- Reorder actions.
- Expand the speed dial menu in a different direction.

# The Proposed Solution

## Add a Label

We implemented the `label` option that specifies the text displayed next to an action button:

```js
$(".action-add").dxSpeedDialAction({
    label: "Add to card",
    icon: "add",
    onClick: function(e) {
        // ...
    }
});
```

## Hide an Action 

To hide a **SpeedDialAction**, you can now set its `visible` option to **false**:

```js
$(".action-add").dxSpeedDialAction({
    icon: "add",
    visible: false,
    onClick: function(e) {
        // ...
    }
});
```

## Reorder Actions

Each **SpeedDialAction** now has an `index` option that specifies the action's position in relation to other actions in the speed dial menu:

```js
$(".action-add").dxSpeedDialAction({
    icon: "add",
    index: 1,
    onClick: function(e) {
        // ...
    }
});

$(".action-edit").dxSpeedDialAction({
    icon: "edit",
    index: 0,
    onClick: function(e) {
        // ...
    }
});
```

## Expand the Menu in a Different Direction

We also added a `direction` option to the `floatingActionButtonConfig`. The option's values (`"up"`, `"down"`, `"auto"`) specify the direction in which the speed dial menu should be expanded:

```js
DevExpress.config({
    floatingActionButtonConfig: {
        position: {
            at: "left top",
            my: "left top",
            offset: "16 16"
        },
        direction: "down"
    },
    // ...
});
```

# Try It

## Live Sandbox

[Angular](https://codesandbox.io/s/overview-devextreme-floating-action-button-h3vy5)

## Installation

Follow the installation guide from the [release page](https://github.com/DevExpress/DevExtreme/releases) and leave us feedback on this topic.

# We Need Your Feedback

## Take a Quick Poll

[Do you find the new Floating Action Button enhancements useful?](https://forms.gle/oHau2BC44cL7TDK69)

# Get Notified of Updates

Subscribe to this thread or to our [Facebook](https://www.facebook.com/DevExpress.DevExtreme/) and [Twitter](https://twitter.com/devextreme) accounts for updates on this topic.

**Comment by figueiredorj** [comment]
Hi,
It would be nice if you add also enabled/disabled property while hands on for speedDialAction.

**Comment by groshenkovamarina** [comment]
Hi @figueiredorj,

This functionality is not in our plans because it is not recommended by the Material guidelines. However, you can use the 'visible' option to hide or show an action in the speed dial menu. Can you provide a scenario where the enabled/disabled property will be useful?

**Comment by groshenkovamarina** [comment]
These features are now available in the [v19.2](https://github.com/DevExpress/DevExtreme/releases/tag/19.2.3) release. I'm closing this thread. In the case of bugs or questions, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
